modules/preview-react/radio/lib/RadioInput.tsx

```diff
@@ -0,0 +1,218 @@
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
+        separation: 2,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    ...mouseFocusBehavior({
+      '&:focus:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:after, &:focus:active:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:checked:after, &:focus:active:checked:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+    }),
+  })
+);
+
+const RadioInputWrapper = styled(Flex)<Pick<RadioLabelProps, 'disabled' | 'variant'>>(
+  {
+    // Hover Ripple element
+    '::before': {
+      content: "''",
+      borderRadius: borderRadius.circle,
+      height: radioHeight,
+      transition: 'box-shadow 150ms ease-out',
+      width: radioWidth,
+      pointerEvents: 'none',
+    },
+  },
+  ({variant, disabled}) => ({
+    '::before': {
+      opacity: variant === 'inverse' ? '0.4' : '1',
+    },
+    '&:hover:before': {
+      boxShadow: disabled ? undefined : `0 0 0 ${rippleRadius}px ${colors.soap200}`,
+    },
+  })
+);
+
+const useRadioInput = createElemPropsHook(useRadioModel)(
+  (model, ref, elemProps: {value?: string; checked?: boolean} = {}) => {
+    const {disabled, variant} = React.useContext(RadioLabelContext);
+    return {
+      'aria-describedby': model.state['aria-describedby'],
+      disabled: disabled,
+      variant: variant,
+      checked: elemProps.value === model.state.value,
+      'aria-checked': elemProps.value === model.state.value,
+      onChange(event: React.ChangeEvent<HTMLInputElement>) {
+        model.onChange(event);
+      },
+      name: model.state.name,
+    };
+  }
+);
+export const RadioInput = createSubcomponent('input')({
+  modelHook: useRadioModel,
+  elemPropsHook: useRadioInput,
+})<RadioLabelProps>(
+  (
+    {
+      children,
+      disabled,
+      checked,
+      value,
+      name,
+      variant,
+      'aria-checked': ariaChecked,
+      'aria-describedby': ariaDescribedby,
+      ...elemProps
+    },
+    Element,
+    model
+  ) => {
+    return (
+      <RadioInputWrapper
+        height="18px"
+        width="18px"
+        flex="0 0 auto"
+        disabled={disabled}
+        variant={variant}
+        {...elemProps}
```

Comment:
We discussed this earlier, but were you able to find a way to avoid spreading `elemProps` to both `RadioInputWrapper` and `StyledRadioInput`? Feels odd to have props spread to both (I would have expected `StyledRadioInput` but not `RadioInputWrapper`.

Reply:
Discussed this with @mannycarrera4 and it sounds like this is needed to ensure the visual tests work properly. Can you add a code comment explaining this for future reference?